DataStructures/Dictionaries/CuckooHashTable.cs

```diff
@@ -164,9 +164,9 @@ private int _cuckooHash(TKey key, int whichHashFunction)
         private bool _isActive(int index)
         {
             if (index < 0 || index > _collection.Length)
-                throw new IndexOutOfRangeException();
+                throw new ArgumentException($"Invalid argument");
```

Comment:
I'm not sure that ArgumentException is beter than IndexOutOfRangeException here.